File: doup-base/src/main/java/com/sleticalboy/base/BasePresenter.java
package com.sleticalboy.base;

import android.content.Context;

/**
 * Created by Android Studio.
 * Date: 1/11/18.
 *
 * @author sleticalboy
 */
public abstract class BasePresenter<V extends IBaseView> {

    private V mView;
    private Context mContext;

    public BasePresenter(Context context) {
        mContext = context;
    }

    protected void setAdapter() {
    }

    protected void setLayoutManager() {
    }

    public void initRecyclerView() {
    }

    protected void onTokenView(V view) {
        mView = view;
    }

    protected void onUnTokenView() {
        mContext = null;
        mView = null;
    }

    public final Context getContext() {
        return mContext;
    }

    public final V getView() {
        return mView;
    }
}

File: doup-base/src/main/java/com/sleticalboy/base/BasePresenter.kt
package com.sleticalboy.base

import android.content.Context

/**
 * Created by Android Studio.
 * Date: 1/11/18.
 *
 * @author sleticalboy
 */
abstract class BasePresenter<V : IBaseView>(context: Context) {

    var view: V? = null
        private set
    var context: Context? = null
        private set

    init {
        this.context = context
    }

    protected open fun setAdapter() {}

    protected open fun setLayoutManager() {}

    protected open fun initRecyclerView() {}

    protected fun onTokenView(view: V) {
        this.view = view
    }

    protected open fun onUnTokenView() {
        context = null
        view = null
    }
}
